__tests__/app.test.js

```diff
@@ -119,87 +119,140 @@ describe("GET /api/articles ", () => {
-          article_id: 5,
-          author: 'icellusedkars',
-          votes: 16,
-          created_at: '2020-06-09T05:00:00.000Z'
-        },
-        {
-          comment_id: 15,
-          body: "I am 100% sure that we're not completely sure.",
-          article_id: 5,
-          author: 'butter_bridge',
-          votes: 1,
-          created_at: '2020-11-24T00:08:00.000Z'
-        }
-      ];
-
-      return request(app)
-        .get("/api/articles/5/comments")
-        .expect(200)
-        .then(({ body }) => {
-          const { comments } = body;
-          expect(comments).toEqual(commentsToCompare);
-          expect(comments).toBeSortedBy("created_at");
-        });
-    });
-
-    test("responds with 200 and an empy array when there are no comments", () => {
-      return request(app)
-        .get("/api/articles/7/comments")
-        .expect(200)
-        .then(({ body }) => {
-          const { comments } = body;
-          expect(comments).toEqual([]);
-        });
-    });
-
-    test("responds with 404 and message article does not exist, when article by id doesnt exist", () => {
-      return request(app)
-        .get("/api/articles/9000/comments")
-        .expect(404)
-        .then(({ body }) => {
-          expect(body.msg).toBe("article does not exist");
-        });
-    });
-
-    test("responds with 400 and message bad request, when given an id that isnt a number", () => {
-      return request(app)
-        .get("/api/articles/apples/comments")
-        .expect(400)
-        .then(({ body }) => {
-          expect(body.msg).toBe("bad request");
-        });
-    });
+        expect(comments).toEqual("This is a great comment!");
+      });
   });
+  test("responds with 404 and message article does not exist, when article by id doesnt exist", () => {
+    const newComment = {
+      username: "Bob Smith",
+      body: "This is a great comment!",
+    };
+
+    return request(app)
+      .post("/api/articles/9000/comments")
+      .send(newComment)
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toBe("article does not exist");
+      });
+  });
+  test("responds with 400 and equest body must include username and body properties if not given a user name", () => {
+    const newComment = {body: 'hello world'};
+
+    return request(app)
+      .post("/api/articles/4/comments")
+      .send(newComment)
+      .expect(400)
+      .then(( {body} ) => {
+        expect(body.msg).toBe('request body must include username and body properties');
+      });
+  });
+
+  test("responds with 404 and User not found if user does not exist", () => {
+    const newComment = { username: 'Bob', body: 'hello world' };
+  
+    return request(app)
+      .post("/api/articles/4/comments")
+      .send(newComment)
+      .expect(404)
+      .then(({ body }) => {
+        console.log(body);
+        expect(body.msg).toBe('User not found');
```

Comment:
You are missing a 400 for an invalid id to post to, eg /api/articles/banana/comments